src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs

```diff
@@ -135,7 +135,7 @@ public async Task ConnectAsync(Uri uri, HttpMessageInvoker? invoker, Cancellatio
                         break;
                     }
                     catch (HttpRequestException ex) when
-                        ((ex.Data.Contains("SETTINGS_ENABLE_CONNECT_PROTOCOL") || ex.Data.Contains("HTTP2_ENABLED"))
+                        ((ex.HttpRequestError == HttpRequestError.ExtendedConnectNotSupported || ex.Data.Contains("HTTP2_ENABLED"))
```

Comment:
```suggestion
                        ((ex.HttpRequestError is  HttpRequestError.ExtendedConnectNotSupported or HttpRequestError.VersionNegotiationError)
```
We should be able to rely on the error code but keep the `HTTP2_ENABLED` in the code for backward compat.

Reply:
Shouldn't it also include `SecureConnectionError`
```suggestion
                        ((ex.HttpRequestError is HttpRequestError.ExtendedConnectNotSupported or HttpRequestError.VersionNegotiationError or HttpRequestError.SecureConnectionError)
```

given that this what we detect in `ConnectHelper.EstablishSslConnectionAsync`:

https://github.com/dotnet/runtime/blob/633d632fb5178245964b41d485cf6f64683657c1/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs#L92C18-L98

Or would that also catch TLS errors we don't want to catch here?

Reply:
Looks like there's an overlap with #75399.

If you specify `Version: 2.0, Policy: Exact` and the server doesn't support HTTP/2, you will get a `SecureConnectionError` error, not a `VersionNegotiationError` with this change.

For H2 WebSockets, we would want to handle that case, but only if HTTP/2 was the problem, not for any TLS error.

I guess we can keep it as-is for now (checking `HTTP2_ENABLED`) and then decide if we can resolve #75399 in a way that makes using the data unnecessary.